src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +32,156 @@ ActorInit Dm_Hina_InitVars = {
+        this->actionFunc = func_80A1F56C;
+    }
+}
+
+void func_80A1F56C(DmHina* this, PlayState* play) {
+    if (Message_GetState(&play->msgCtx) == (u8)2) {
+        this->unk17C = 2;
+        this->actionFunc = func_80A1F5AC;
+    }
+}
+
+void func_80A1F5AC(DmHina* this, PlayState* play) {
+    this->unk17C--;
+    if (this->unk17C == 0) {
+        this->unk180 = 1;
+        Cutscene_StartManual(play, &play->csCtx);
+        this->unk17A = Play_CreateSubCamera(play);
+        Play_ChangeCameraStatus(play, 0, 1);
+        Play_ChangeCameraStatus(play, this->unk17A, 7);
+        this->actionFunc = func_80A1F63C;
+    }
+}
+
+void func_80A1F63C(DmHina* this, PlayState* play) {
+    this->unk160 = this->actor.world.pos.x + 100.0f;
+    this->unk164 = this->unk_158 + 260.0f;
+    this->unk168 = this->actor.world.pos.z + 100.0f;
+    this->unk16C = this->actor.world.pos.x;
+    this->unk174 = this->actor.world.pos.z;
+    this->unk170 = this->actor.world.pos.y + (this->unk154 * this->unk_15C) + (40.0f * this->unk_15C);
+    Play_SetCameraAtEye(play, this->unk17A, (Vec3f*)&this->unk16C, (Vec3f*)&this->unk160);
+    Math_SmoothStepToF(&this->actor.world.pos.y, this->unk_158 + 300.0f, 0.5f, 2.0f, 0.1f);
+    if (((this->unk_158 + 240.0f) < this->actor.world.pos.y) && (this->unk17E != 1)) {
+        this->unk17E = 1;
+        Actor_PlaySfx(&this->actor, 0x580BU);
+    }
+    Actor_PlaySfx(&this->actor, 0x219BU);
+}
+
+void func_80A1F75C(DmHina* this, PlayState* play) {
+    s32 var_a0;
+
+    switch (this->unk17E) {
+        case 0:
+            this->unk17F = 0;
+            this->unk14C = 0.0f;
+            break;
+        case 1:
+            Math_SmoothStepToF(&this->unk14C, 1.0f, 0.4f, 0.05f, 0.001f);
+            this->unk17F = this->unk14C * 255.0f;
+            this->unk150 = Math_SinS(play->state.frames * 0x1F40);
+            for (var_a0 = 0; var_a0 < 3; var_a0++) {
+                play->envCtx.lightSettings.ambientColor[var_a0] = play->envCtx.lightSettings.fogColor[var_a0] =
+                    play->envCtx.lightSettings.diffuseColor1[var_a0] = -255.0f * this->unk14C;
+            }
+            play->envCtx.lightSettings.fogNear = -500.0f * this->unk14C;
+            if (play->envCtx.lightSettings.fogNear < -300) {
+                play->roomCtx.curRoom.segment = NULL;
+            }
+            break;
+    }
+}
+
+void DmHina_Update(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->actionFunc(this, play);
+    func_80A1F75C(this, play);
+}
+
+extern UNK_TYPE D_04023348;
+extern UNK_TYPE D_04023428;
+
+void func_80A1F9AC(DmHina* this, PlayState* play) {
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+
+    if (this->unk17E != 0) {
+        OPEN_DISPS(gfxCtx);
+
+        Gfx_SetupDL25_Xlu(play->state.gfxCtx);
+
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, (u8)(this->unk14C * 100.0f) + 0x9B, this->unk17F);
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, (u8)(this->unk150 * 100.0f) + 0x32, 0);
```

Comment:
```suggestion
        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, (u8)(this->unk14C * 100.0f) + 155, this->unk17F);
        gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, (u8)(this->unk150 * 100.0f) + 50, 0);
```
We prefer decimal for colors.